### PR TITLE
Add denormalize_relations configuration

### PR DESCRIPTION
**Pull request by odolbeau**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Far from perfect:
* I didn't try with circular references, not sure it will be handled correctly.
* There is no tests. It's linked to #306 I guess but I have no idea of how to run a behat tests with another configuration. :/
* Only 1 configuration entry for both normalization / denormalization, is it a good idea?

At least, it works for my use case, that's better than nothing. :)

### CODE REVIEW

**Comment by sroze** [comment]
Could you add tests?

**Comment by odolbeau** [comment]
First attempt to add tests.
I still need to create test dedicated to this new option but I would like to have your opinion on this implementation first.

File: features/fixtures/TestAppNormalizeRelations/config/config.yml [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    description:                       "This is a test API."
+    collection:
+        filter_name:
+            order:                     "order"
+        order:                         "ASC"
+        pagination:
+            items_per_page:
+                number:                3
+                enable_client_request: true
+    denormalize_relations: false
```
Comment (odolbeau) [inline features/fixtures/TestAppNormalizeRelations/config/config.yml]: The only difference with the other config.yml file is this line... :/

**Comment by odolbeau** [comment]
screen shot 2015-10-25 at 14 29 02
Here you can see the diff between `features/suites/denormalize_relations/hydra/doc.feature` & `features/suites/normalize_relations/hydra/doc.feature`

File: features/suites/normalize_relations/hydra/error.feature [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    When I send a "POST" request to "/dummies" with body:
+    """
+    {
+      "name": "Foo",
+      "relatedDummy": {
+        "name": "bar"
+      }
+    }
+    """
+    Then the response status code should be 500
```
Comment (odolbeau) [inline features/suites/normalize_relations/hydra/error.feature]: It's a doctrine exception of type "A new entity was found through the relationship...".
I don't know how to deal with this exception. There is the same one without `denormalize_relations` to false if you don't configure cascade persist correctly.
For me it's not a problem to have & 500 here. In fact, I'm not even sure it should be under test.

**Comment by dunglas** [comment]
What is the interest of adding this configuration option instead of using the (already supported) Serializer groups?

**Comment by odolbeau** [comment]
Cause I don't want to add a group on all properties of all my entities... :/
Do you see any other way to have this results without adding a new option which will probably not be used a lot?

**Comment by dunglas** [comment]
I think you can use a custom `Normalizer` class decorating bundle normalizers and fitting your project needs.

**Pull request closed without merging**